pancancer-arch-3/src/main/java/info/pancancer/arch3/reporting/SlackRenderer.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2015 CancerCollaboratory
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package info.pancancer.arch3.reporting;
+
+import com.ullink.slack.simpleslackapi.SlackAttachment;
+import info.pancancer.arch3.beans.JobState;
+import info.pancancer.arch3.beans.ProvisionState;
+import info.pancancer.arch3.beans.Status;
+import info.pancancer.arch3.reporting.ReportAPI.Commands;
+import io.cloudbindle.youxia.listing.AbstractInstanceListing;
+import java.util.Date;
+import java.util.Map;
+import java.util.Map.Entry;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * THe SlackRenderer takes generic data from the report API and formats it for output to Slack.
+ *
+ * @author dyuen
+ */
+public class SlackRenderer {
+    private final ReportAPI reportAPI;
+
+    public SlackRenderer(ReportAPI reportAPI) {
+        this.reportAPI = reportAPI;
+    }
+
+    public FormattedMessage convertToResult(String message) {
+        if (message.startsWith(Commands.STATUS.name())) {
```

Comment:
I'm not sure how "message" is formatted, but if there's a sensible way to extract the leading string that indicates what to do next, these if-else-if... could probably be cleaned up with switch/case ... not critical, but would probably improve readbility.